src/olympia/reviewers/utils.py

```diff
@@ -322,13 +322,27 @@ def get_actions(self, request):
             request, amo.permissions.REVIEWS_ADMIN)
         reviewable_because_complete = self.addon.status not in (
             amo.STATUS_NULL, amo.STATUS_DELETED)
+        regular_addon_review_is_allowed = (
+            not self.content_review_only and
+            not self.addon.needs_admin_code_review and
+            not self.addon.needs_admin_content_review and
+            not self.addon.needs_admin_theme_review
+        )
+        regular_content_review_is_allowed = (
+            self.content_review_only and
+            not self.addon.needs_admin_content_review and
+            (
+                not self.addon.needs_admin_code_review or
+                self.version.source
```

Comment:
Still a little concerned that a version that has a source uploaded may need admin review for something else but I guess you're aware of the trade-off.

Reply:
This is why we are checking whether source code has been attached. If it is in admin review it might or might not have sources attached regardless and we are aware this is a bit fuzzy, but we cannot solve it without major refactoring, e.g. storing *why* the admin flag was set.

Reply:
looking forward to the followup pr that adds a fourth type of admin flag 😆

Reply:
🤯